src/github.com/matrix-org/dendrite/roomserver/query/query.go

```diff
@@ -418,6 +418,108 @@ func (r *RoomserverQueryAPI) QueryServerAllowedToSeeEvent(
 	return nil
 }
 
+// QueryStateAndAuthChain implements api.RoomserverQueryAPI
+func (r *RoomserverQueryAPI) QueryStateAndAuthChain(
```

Comment:
There's a lot of code here for there to be no tests at all. Let's at least have some simple test cases so that we're more inclined to add tests in the future.

Reply:
Added tests.